compiler/resolution/functionResolution.cpp

```diff
@@ -7601,6 +7639,17 @@ static void instantiate_default_constructor(FnSymbol* fn) {
         }
       }
     }
+
+    if (AggregateType* ct = toAggregateType(fn->_this->type)) {
+      if (ct->instantiationStyle == DEFINES_INITIALIZER &&
```

Comment:
As an aside, buildDefaultWrapper does a lot of "stuff" for cases where arrays depend on the type of a domain. E.g.

   record R { var D:domain(1); var A:[D] int; }

I'm pretty unsatisfied with the way this works now. I just want to point out that it's complicated!